src/components/shared/Entity/ExistingEntityCards/Store/create.ts

```diff
@@ -0,0 +1,100 @@
+import { getLiveSpecsByConnectorId } from 'api/liveSpecsExt';
+import { ExistingEntityState } from 'components/shared/Entity/ExistingEntityCards/Store/types';
+import produce from 'immer';
+import { getStoreWithHydrationSettings } from 'stores/Hydration';
+import { ExistingEntityStoreNames } from 'stores/names';
+import { devtoolsOptions } from 'utils/store-utils';
+import { create, StoreApi } from 'zustand';
+import { devtools, NamedSet } from 'zustand/middleware';
+
+const getInitialStateData = (): Pick<
+    ExistingEntityState,
+    'connectorName' | 'createNewTask' | 'queryData'
+> => ({
+    connectorName: null,
+    createNewTask: false,
+    queryData: null,
+});
+
+const getInitialState = (
+    set: NamedSet<ExistingEntityState>,
+    get: StoreApi<ExistingEntityState>['getState']
+): ExistingEntityState => {
+    return {
+        ...getInitialStateData(),
+        ...getStoreWithHydrationSettings('Existing Entity Store', set),
+
+        setQueryData: (value) => {
+            set(
+                produce((state: ExistingEntityState) => {
+                    state.queryData = value;
+                }),
+                false,
+                'Query Data Set'
+            );
+        },
+
+        setConnectorName: (value) => {
+            set(
+                produce((state: ExistingEntityState) => {
+                    state.connectorName = value;
+                }),
+                false,
+                'Connector Name Set'
+            );
+        },
+
+        setCreateNewTask: (value) => {
+            set(
+                produce((state: ExistingEntityState) => {
+                    state.createNewTask = value;
+                }),
+                false,
+                'Create New Task Flag Set'
+            );
+        },
+
+        hydrateState: async (entityType, connectorId) => {
+            if (connectorId) {
+                const { setCreateNewTask, setHydrationErrorsExist } = get();
+
+                const { data, error } = await getLiveSpecsByConnectorId(
+                    entityType,
+                    connectorId
+                );
+
+                if (error) {
+                    setHydrationErrorsExist(true);
+                    setCreateNewTask(true);
```

Comment:
We should `return` here

Reply:
I feel combining the this if statement with the if/else statement below would make more sense than throwing in a return statement.

```ts
if (error) {
    ...
} else if (data && data.length > 0) {
    ...
} else {
    ...
}
```